remote/remote_state_s3.go

```diff
@@ -117,8 +126,28 @@ func parseS3Config(config map[string]interface{}) (*RemoteStateConfigS3, error)
 	return &s3Config, nil
 }
 
+// Parse the given map into an extended S3 config
+func parseExtendedS3Config(config map[string]interface{}) (*ExtendedRemoteStateConfigS3, error) {
+	var s3Config RemoteStateConfigS3
+	var extendedConfig ExtendedRemoteStateConfigS3
+
+	if err := mapstructure.Decode(config, &s3Config); err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	if err := mapstructure.Decode(config, &extendedConfig); err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
```

Comment:
Isn't it enough to just do this? Doesn't it automatically fill in the `remoteStateConfigS3` portion?

Reply:
No, that doesn't work. It does create the extendedConfig.remoteStateConfigS3 object, but all the fields are empty.